### ISSUE TITLE
Duplicated class errors

### ISSUE THREAD

**Issue by adam12** [soutaro/steep#89]
Hello,

I was just wondering if it's possible to define signatures for duplicate classes (mostly ones already defined in Steep for stdlib).

Example:

```ruby
# types.rbi
class Module
  def some_method: -> void
end
```

```ruby
# some_method.rb
class Module
  def some_method; end
end
```

```
$ steep check -I types.rbi some_method.rb
#<RuntimeError: Duplicated class: ::Module>
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/ast/signature/env.rb:31:in `add'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project.rb:131:in `block (3 levels) in reload_signature'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project.rb:130:in `each'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project.rb:130:in `block (2 levels) in reload_signature'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project.rb:125:in `each_value'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project.rb:125:in `block in reload_signature'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project/listener.rb:35:in `load_signature'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project.rb:124:in `reload_signature'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project.rb:55:in `block in type_check'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project/listener.rb:13:in `check'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/project.rb:53:in `type_check'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/drivers/check.rb:56:in `run'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/cli.rb:231:in `block in process_check'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/cli.rb:199:in `with_signature_options'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/cli.rb:209:in `process_check'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/lib/steep/cli.rb:166:in `run'
  /usr/home/adam/.rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/steep-0.11.0/exe/steep:11:in `<top (required)>'
  /home/adam/.rubies/latest/bin/steep:23:in `load'
  /home/adam/.rubies/latest/bin/steep:23:in `<main>'
```

**Comment by soutaro** [comment]
Hi,

Using `extension` construct in signatures is the way we propose.

```
extension Module (SomeMethod)
  def some_method: () -> void
end
```

One problem left is you will see many `MethodDefinitionMissing` errors in the module definition if you use `--strict` mode, which helps to find a method you forget to implement.

```
some_method.rb:1:0: MethodDefinitionMissing: module=::Module, method=attr_reader (class Module)
...
```

I will fix the problem after finished #88.

**Comment by adam12** [comment]
Worked great. Thank you for the prompt response :)
